### PR TITLE
Add/fix some telemetry for driver around epoch tracker

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Add/fix some telemetry for driver around epoch.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/epochTracker.ts [hunk 1]
```diff
@@ -453,7 +453,7 @@ export class EpochTracker implements IPersistedFileCache {
 			return new NonRetryableError(
 				"Epoch mismatch",
 				DriverErrorType.fileOverwrittenInStorage,
-				{ driverVersion },
+				{ driverVersion, serverEpoch: epochFromResponse },
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/epochTracker.ts:456]: Should we also add clientEpoch: this.fluidEpoch here (and remove it from checkForEpochError)? The code is hard to follow though

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 2]
```diff
@@ -603,8 +603,17 @@ export class DocumentDeltaConnection
 			// Socket can be disconnected while waiting for Fluid protocol messages
 			// (connect_document_error / connect_document_success), as well as before DeltaManager
 			// had a chance to register its handlers.
-			this.addTrackedListener("disconnect", (reason) => {
-				const err = this.createErrorObject("disconnect", reason);
+			this.addTrackedListener("disconnect", (reason, description) => {
+				const err = this.createErrorObject(
+					"disconnect",
+					`${reason}${
+						typeof description === "object"
```
Comment (vladsud) [inline packages/drivers/driver-base/src/documentDeltaConnection.ts:610]: Can you please add reference here to https://socket.io/docs/v4/client-api/#event-disconnect ?
I do not see documentation telling anything about `description` shape.
Also we should avoid using instanceof if possible, as it does not work for objects created in another iframe (unlikely to be here, but in general - you never know where objects might be created).
Can we instead use createErrorObject() with this object?

Here is an example of what we get for websocket errors:
socket.io (connect_error): websocket error: {"description":{"isTrusted":true},"type":"TransportError"}

Comment (jatgarg) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:610]: Actually I looked into socket.io code in their repo and seems like they never add this arg. So I am removing this.

**Review by vladsud — APPROVED** [review]
